### PR TITLE
Replace apache http client with airlift

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java [hunk 1]
```diff
@@ -43,6 +42,8 @@ public class ClusterStatsHttpMonitor
 {
     private static final Logger log = LoggerFactory.getLogger(ClusterStatsHttpMonitor.class);
     private static final String SESSION_USER = "sessionUser";
+    private static final int HTTP_OK_CODE = 200;
```
Comment (mosabua) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:45]: Remove

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java [hunk 2]
```diff
@@ -151,9 +152,9 @@ private String queryCluster(ProxyBackendConfiguration backend, String path)
 
         try (Response res = call.execute()) {
             switch (res.code()) {
-                case HttpStatus.SC_OK:
+                case HTTP_OK_CODE:
                     return res.body().string();
```
Comment (mosabua) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:156]: use 

HttpStatus.OK_200

from Jetty instead

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java [hunk 3]
```diff
@@ -151,9 +152,9 @@ private String queryCluster(ProxyBackendConfiguration backend, String path)
...
-                case HttpStatus.SC_UNAUTHORIZED:
+                case HTTP_UNAUTHORIZED_CODE:
```
Comment (mosabua) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:157]: use HttpStatus.UNAUTHORIZED_401

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java (hunk 1 above)
Comment (ebyhr) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:45]: This constant is used in L155.

File: gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyHandler.java [hunk 4]
```diff
@@ -105,7 +104,7 @@ protected void debugLogHeaders(Request proxyRequest)
 
     protected boolean isGZipEncoding(HttpServletResponse response)
     {
-        String contentEncoding = response.getHeader(HttpHeaders.CONTENT_ENCODING);
+        String contentEncoding = response.getHeader("Content-Encoding");
```
Comment (mosabua) [inline gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyHandler.java:107]: Use HttpHeader.CONTENT_ENCODING from jetty

**Review by mosabua — CHANGES_REQUESTED** [review]

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java (hunk 1 above)
Comment (mosabua) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:45]: See below... both are not needed.

**Review by mosabua — APPROVED** [review]
Looks good to me now. Thank you @ebyhr

**Pull request merged**